datasetinsights/io/downloader/unity_simulation.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+import re
+from pathlib import Path
+
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.usim import Downloader, download_manifest
+
+logger = logging.getLogger(__name__)
+
+
+class UnitySimulationDownloader(DatasetDownloader):
+    """ Downloads unity simulation datasets
+        Args:
+        access_token: "Access token to be used to authenticate to
+         unity simulation for downloading the dataset"
+
+    """
+
+    PROTOCOL = "usim://"
+
+    def __init__(self, access_token=None):
```

Comment:
Although not supported to override source_uri, we should allow `run_execution_id` and `project_id` to be part of the arguments in the `__init__` method and default to `None`. Also, this should accept `**kwargs` to capture unwanted parameters.

Reply:
I agree with accepting **kwargs in the constructor. Ii don't think it makes sense to have run_execution_id and project_id be a part of the constructor. If anything they could be passed to the download method with default=None. The reason being it does not make sense to reinitialize the Downloader every time a new dataset has to be downloaded.

Reply:
In this case, I think it makes sense not to have instance variable for run-execution-id and project_id. These two parameters should be input of `UnitySimulationDatasetDownloader.download` method, as when the Downloader is initialized it only have required connection/auth info.